src/cleaning/clean.jl

```diff
@@ -0,0 +1,43 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    clean(mesh)
+
+Remove the unused vertices of `mesh`.
+"""
+function clean(mesh)
+  # get the faces of the mesh
+  topo = convert(HalfEdgeTopology, topology(mesh))
+  ∂₂₀ = Boundary{2,0}(topo)
+  nfaces = nelements(topo)
+
+  # vector to store the indices of the used vertices
+  used = Vector{Int}(undef, 0)
+
+  # dictionary to map the old indices to the new indices
+  newindices = Dict{Int,Int}()
+
+  # vector to store the connectivities of the cleaned mesh
+  connectivities = Vector{Connectivity}(undef, 0)
+
+  # l=length(used) will be incremented
+  l = 0
+
+  # iterate over faces
+  for f in 1:nfaces
+    face = ∂₂₀(f)
```

Comment:
this is not a face, please use a more descriptive name like inds to help the reader of the code that this variable stores a list of indices